source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/ActionLayoutRenderer.java

```diff
@@ -110,33 +106,82 @@ else if (alignment == ActionAlignment.Center.swigValue())
             }
         }
 
-        int i = 0;
         long maxActions = hostConfig.GetActions().getMaxActions();
 
         // Allow the actions to have the icon drawn at the top as long as all actions have an icon
         renderArgs.setAllowAboveTitleIconPlacement(true);
-        for(; i < size && i < maxActions; ++i)
+
+        //Splits "Primary" and "Secondary" actions.
+        BaseActionElementVector primaryActionElementVector = new BaseActionElementVector();
+        BaseActionElementVector secondaryActionElementVector = new BaseActionElementVector();
+
+        for (BaseActionElement actionElement : baseActionElementList)
         {
-            BaseActionElement actionElement = baseActionElementList.get(i);
-            if(actionElement.GetIconUrl().isEmpty())
+            if (actionElement.GetMode() == Mode.Secondary)
+            {
+                secondaryActionElementVector.add(actionElement);
+            }
+            else
+            {
+                primaryActionElementVector.add(actionElement);
+            }
+
+            if (actionElement.GetIconUrl().isEmpty())
             {
                 renderArgs.setAllowAboveTitleIconPlacement(false);
-                break;
             }
         }
 
-        if (i >= maxActions && size != maxActions)
+        int primaryElementsSize = primaryActionElementVector.size();
+        if (primaryElementsSize > maxActions)
+        {
+            List<BaseActionElement> excessElements = primaryActionElementVector.subList((int) maxActions, primaryElementsSize);
+            //Add excess elements to the secondary list if flag is enabled.
+            if (hostConfig.GetActions().getAllowMoreThanMaxActionsInOverflowMenu())
```

Comment:
The implementation in JS was reviewed and agreed upon during a design discussion meeting that Robin Liao was attending so that she'd know what to do. Robin and I finalized the name of the flag later when she submitted her initial PR.

Reply:
Removed. As discussed with @robin-liao moved to client callback.